base/sparse/sparsevector.jl

```diff
@@ -17,6 +18,9 @@ typealias BinaryOp Union{Function, Func{2}}
 
 ### Types
 
+# Assumes that no zeros are stored in nzval
```

Comment:
explicit stored zeros should be allowed - I don't think anything would break down if that happens